carma/scripts/pose_offset_node.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2022 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import rospy
+from geometry_msgs.msg import PoseStamped
+from geometry_msgs.msg import Vector3
+
+# Worker class for core business logic
+class PoseOffsetWorker:
+    def __init__(self, pub, x_offset, y_offset, z_offset):
+        self.pub = pub
+        self.x_offset = x_offset
+        self.y_offset = y_offset
+        self.z_offset = z_offset
+
+    # Callback to apply offset based on parameters
+    def pose_cb(self, msg):
+      msg.pose.position.x += self.x_offset
+      msg.pose.position.y += self.y_offset
+      msg.pose.position.z += self.z_offset
+
+      self.pub.publish(msg)
+
+    # Method to update parameters at runtime based on topic input
+    def param_cb(self, msg):
+      self.x_offset = msg.x
+      self.y_offset = msg.y
+      self.z_offset = msg.z
+
+
+
+
+def run():
+
+  rospy.init_node('pose_offset_node')
+
+  # Load params
+  x_offset = rospy.get_param("/pose_x_offset", 1.0)
+  y_offset = rospy.get_param("/pose_y_offset", 1.0)
+  z_offset = rospy.get_param('/pose_z_offset', 0.0)
+
+  # Setup publishers
+  pub = rospy.Publisher('selected_pose', PoseStamped, queue_size=10)
```

Comment:
This PR contains multiple changes (such as this file) which were explicitly asked to be removed in this PR
https://github.com/usdot-fhwa-stol/carma-platform/pull/1983

Why are they back? Please clean up the commits on this branch.